### ISSUE TITLE
Component: Dropdown

### ISSUE THREAD

**Issue by KRONTri** [primefaces/primereact#4560]
### Describe the bug

Hi

we're creating a dynamic form from JsonSchema.
The values of all form elements are fetched from our backend. 

In case the schema has an enum property set we create a `<Dropdown>` and fill it with those enum values.

If the enum consists of string values like `["a","b","c"] this works flawless.
If the enum consists of integer values like `[0,1,2]` it doesn't work completely.

Value fetched from backend for example is 0 (integer). 
If we fill value 0 into the `<Dropdown>` it is visibly selected in expanded state but the Dropdown itself doesn't show the value but shows the "Choose item" placeholder instead.

In the documentation `<Dropdown>` value can be `any`.

Best regards
KRON

### Reproducer

https://stackblitz.com/edit/vitejs-vite-hb1pme?file=src%2FApp.tsx

### PrimeReact version

9.5.0

### React version

18.x

### Language

ES6

### Build / Runtime

Vite

### Steps to reproduce the behavior

1. Create `<Dropdown>` with `options={[0,1]}` and `value={0}`
2. Dropdown shows placeholder but if you expand the dropdown the value 0 is selected.

### Expected behavior

Value 0 is shown instead of placeholder

**Comment by ubau** [comment]
This is working example:
```js
function App() {
  const [value, setValue] = useState(0);
  return (
    <div className="flex flex-column">
      <Dropdown options={[0, 1, 2]} value={value} onChange={(e) => setValue(e.value)} />
    </div>
  );
}
```

**Comment by melloware** [comment]
I had fixed this in 7.0 but looks like its cropped back up see this ticket: https://github.com/primefaces/primereact/issues/2349

**Comment by melloware** [comment]
@ubau you are right the component is not being used correctlty in the example but is in yours!

**Comment by melloware** [comment]
@ubau actually its still showing as "empty" in your example: https://stackblitz.com/edit/vitejs-vite-72n3ey

**Comment by ubau** [comment]
if you want 0 instead of "empty" add ` valueTemplate={selectedTemplate}` to `Dropdown` and
```js
const selectedTemplate  =  (option: number)  =>  {
  return (<div>{option}</div>)
}
```
to current component

**Comment by melloware** [comment]
I am not sure it does seem weird your can't pick 0.  What if it was a database ID or something as 0?

**Comment by ubau** [comment]
@melloware you can pick 0, it's just shows as "empty" by default because it is falsy value. To show it as 0 you need template as above.

**Comment by KRONTri** [comment]
Then it should be documented somewhere that falsy values will display as "empty". Maybe I'm just blind.
I implemented the `valueTemplate`  and it works. 

Thanks.

**Comment by ubau** [comment]
You should think twice and double check if you want use falsy values, somewhere else it can show something else or show nothing. It is general knowledge.
